lib/blockchain_double.js

```diff
@@ -503,30 +514,31 @@ BlockchainDouble.prototype.processCall = function(tx, blockNumber, callback) {
 
       // We checkpoint here for speed. We want all state trie reads/writes to happen in memory,
       // and the final output be flushed to the database at the end of transaction processing.
-      self.vm.stateManager.checkpoint();
-
-      var runArgs = {
-        tx: tx,
-        block: block,
-        skipBalance: true,
-        skipNonce: true
-      };
-
-      self.vm.runTx(runArgs, function(vmerr, result) {
-        // This is a check that has been in there for awhile. I'm unsure if it's required, but it can't hurt.
-        if (vmerr && vmerr instanceof Error === false) {
-          vmerr = new Error("VM error: " + vmerr);
-        }
+      self.vm.stateManager._cache.clear();
```

Comment:
hmmmm why is this necessary?

Reply:
REASONS, SEESE. REASONS!

When we update the stateRoot to the stateRoot of a different block the data in the cache isn't updated with it. If, for instance, an account is in the cache from a previous call to access the account from the `latest` block, but we are at this time in the code operating in the context of an older block where the account may have different values, `getAccount` will return the wrong values, despite the `stateRoot` being updated to the correct block.

I do think we need to rework our implementation of the `vm` and its `stateManager` in the near future to avoid these kinds of shenanigans.